diff -Nru sawfish-0.30/configure.in sawfish-0.30.new/configure.in
--- sawfish-0.30/configure.in	Thu Jul 13 04:07:04 2000
+++ sawfish-0.30.new/configure.in	Thu Jul 13 04:06:22 2000
@@ -55,10 +55,10 @@
 esac
 
 sawfishdir='${datadir}/sawfish'
-lispdir='${sawfishdir}/${version}/lisp'
-nokogiridir='${sawfishdir}/${version}/nokogiri'
-localedir='${sawfishdir}/${version}/locale'
-sawfishexecdir='${libexecdir}/sawfish/${version}/${host_type}'
+lispdir='${sawfishdir}/lisp'
+nokogiridir='${sawfishdir}/nokogiri'
+localedir='$(datadir)/locale'
+sawfishexecdir='${libexecdir}/sawfish/'
 
 dnl Checks for programs.
 AC_PROG_CC
diff -Nru sawfish-0.30/sounds/Makefile.in sawfish-0.30.new/sounds/Makefile.in
--- sawfish-0.30/sounds/Makefile.in	Thu Jul 13 04:07:04 2000
+++ sawfish-0.30.new/sounds/Makefile.in	Thu Jul 13 04:06:22 2000
@@ -27,7 +27,7 @@
 VPATH=@srcdir@:@top_srcdir@
 
 sitesoundsdir := ${sawfishdir}/sounds
-soundsdir := ${sawfishdir}/${version}/sounds
+soundsdir := ${sawfishdir}/sounds
 
 all :
 
diff -Nru sawfish-0.30/themer/Makefile.in sawfish-0.30.new/themer/Makefile.in
--- sawfish-0.30/themer/Makefile.in	Thu Jul 13 04:07:04 2000
+++ sawfish-0.30.new/themer/Makefile.in	Thu Jul 13 04:09:13 2000
@@ -26,8 +26,7 @@
 srcdir=@srcdir@
 VPATH=@srcdir@:@top_srcdir@
 
-SUBSTS := -e "s|__version__|$(version)|" \
-	  -e "s|__sawfishdir__|$(sawfishdir)|" \
+SUBSTS := -e "s|__sawfishdir__|$(sawfishdir)|" \
 	  -e "s|__locale_dir__|$(localedir)|"
 
 all : themer.jlc
@@ -40,14 +39,14 @@
 
 install : all installdirs
 	$(INSTALL_SCRIPT) themer.jlc $(DESTDIR)$(bindir)/sawfish-themer
-	$(INSTALL_DATA) $(srcdir)/themer.glade $(DESTDIR)$(sawfishdir)/$(version)
+	$(INSTALL_DATA) $(srcdir)/themer.glade $(DESTDIR)$(sawfishdir)
 
 uninstall :
 	rm -f $(DESTDIR)${bindir}/sawfish-themer
-	rm -f $(DESTDIR)$(sawfishdir)/$(version)/themer.glade
+	rm -f $(DESTDIR)$(sawfishdir)/themer.glade
 
 installdirs : mkinstalldirs
-	$(SHELL) $< $(DESTDIR)$(bindir) $(DESTDIR)$(sawfishdir)/$(version)
+	$(SHELL) $< $(DESTDIR)$(bindir) $(DESTDIR)$(sawfishdir)
 
 clean :
 	rm -f *~ themer.jl *.jlc *.bak core
diff -Nru sawfish-0.30/themer/themer.in sawfish-0.30.new/themer/themer.in
--- sawfish-0.30/themer/themer.in	Thu Jul 13 04:07:04 2000
+++ sawfish-0.30.new/themer/themer.in	Thu Jul 13 04:06:22 2000
@@ -31,7 +31,7 @@
 (setq backtrace-on-error '(bad-arg missing-arg void-value invalid-function))
 
 (defvar version "__version__")
-(defvar glade-file "__sawfishdir__/__version__/themer.glade")
+(defvar glade-file "__sawfishdir__/themer.glade")
 (defvar locale-dir "__locale_dir__")
 
 (when (and (not (get-command-line-option "--disable-nls"))
diff -Nru sawfish-0.30/themes/Makefile.in sawfish-0.30.new/themes/Makefile.in
--- sawfish-0.30/themes/Makefile.in	Thu Jul 13 04:07:04 2000
+++ sawfish-0.30.new/themes/Makefile.in	Thu Jul 13 04:06:22 2000
@@ -27,7 +27,7 @@
 VPATH=@srcdir@:@top_srcdir@
 
 sitethemedir := ${sawfishdir}/themes
-themedir := ${sawfishdir}/${version}/themes
+themedir := ${sawfishdir}/themes
 
 # also in lisp/Makefile.in
 sawfish_prog = $(rep_LIBTOOL) --mode=execute -dlopen ../src/gradient.la \
