diff -urN sawfish-0.38.org/capplet/Makefile.in sawfish-0.38/capplet/Makefile.in
--- sawfish-0.38.org/capplet/Makefile.in	Sat Mar 24 20:32:33 2001
+++ sawfish-0.38/capplet/Makefile.in	Sat Mar 24 20:33:06 2001
@@ -26,8 +26,8 @@
 srcdir=@srcdir@
 VPATH=@srcdir@:@top_srcdir@
 
-override CFLAGS += $(GNOME_CC_CFLAGS)
-override LIBS += $(GNOME_CC_LIBS)
+override CFLAGS += $(GNOME_CC_CFLAGS) $(GTK_CFLAGS)
+override LIBS += $(GNOME_CC_LIBS) $(GTK_LIBS)
 
 G_CC_DIR = $(GNOMEDIR)/share/control-center
 G_MENU_DIR = $(GNOMEDIR)/share/gnome/apps/Settings
