--- sawfish-1.0/src/Makefile.in~	Wed Jun 20 23:43:16 2001
+++ sawfish-1.0/src/Makefile.in	Wed Sep  5 19:33:01 2001
@@ -25,6 +25,7 @@
 top_srcdir=@top_srcdir@
 srcdir=@srcdir@
 VPATH=@srcdir@:@top_srcdir@
+LIBTOOL = @LIBTOOL@
 
 SRCS :=	colors.c cursors.c display.c events.c fonts.c frames.c \
 	functions.c images.c keys.c main.c multihead.c pixmap-cache.c \
@@ -43,7 +44,7 @@
 all : sawfish libclient.o $(DL_OBJS) .libexec gtk-style
 
 sawfish : $(OBJS) $(LIBOBJS)
-	$(rep_LIBTOOL) --mode=link $(CC) -export-dynamic $(LDFLAGS) \
+	$(LIBTOOL) --mode=link $(CC) -export-dynamic $(LDFLAGS) \
 	  -o sawfish $(OBJS) $(LIBOBJS) $(REP_LIBS) \
 	  $(IMAGE_LIBS) $(X11_LIBS) $(EXTRA_X11_LIBS) $(LIBS)
 
@@ -54,7 +55,7 @@
 	$(rep_DL_LD) $(LDFLAGS) -o $@ $< $(ESD_LIBS) $(AUDIOFILE_LIBS)
 
 libclient_.lo : libclient.c
-	$(rep_LIBTOOL) --mode=compile $(CC) -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
+	$(LIBTOOL) --mode=compile $(CC) -c $(CPPFLAGS) $(CFLAGS) -o $@ $<
 
 client.la : client.lo libclient_.lo
 	$(rep_DL_LD) $(LDFLAGS) -o $@ $^ $(X11_LIBS) $(REP_LIBS) $(LIBS)
@@ -64,23 +65,23 @@
 
 install : all installdirs
 	for p in sawfish; do \
-	  $(rep_LIBTOOL) --mode=install $(INSTALL_PROGRAM) $$p $(DESTDIR)${bindir}; \
+	  $(LIBTOOL) --mode=install $(INSTALL_PROGRAM) $$p $(DESTDIR)${bindir}; \
 	done
 	$(rep_DL_INSTALL) gtk-style $(DESTDIR)${sawfishexecdir}
 	$(foreach x,$(DL_DSTS),\
-	  $(rep_LIBTOOL) --mode=install $(INSTALL_PROGRAM) \
+	  $(LIBTOOL) --mode=install $(INSTALL_PROGRAM) \
 	  $(notdir $(x)) $(DESTDIR)$(sawfishexecdir)/$(dir $(x));)
 	$(rep_INSTALL_ALIASES) -c . $(DESTDIR)$(sawfishexecdir)
-	$(rep_LIBTOOL) --mode=install $(INSTALL_PROGRAM) client.la $(DESTDIR)$(repexecdir)/sawfish/
+	$(LIBTOOL) --mode=install $(INSTALL_PROGRAM) client.la $(DESTDIR)$(repexecdir)/sawfish/
 
 uninstall :
 	for p in sawfish; do \
-	  $(rep_LIBTOOL) rm $(DESTDIR)${bindir}/$$p; \
+	  $(LIBTOOL) rm $(DESTDIR)${bindir}/$$p; \
 	done
 	rm -f $(DESTDIR)$(bindir)/sawfish-client
 	$(rep_DL_UNINSTALL) $(DESTDIR)${sawfishexecdir}/gtk-style
 	for dl in $(DL_DSTS); do \
-	  $(rep_LIBTOOL) rm $(DESTDIR)${sawfishexecdir}/$$dl; \
+	  $(LIBTOOL) rm $(DESTDIR)${sawfishexecdir}/$$dl; \
 	done
 	$(rep_DL_UNINSTALL) $(DESTDIR)$(repexecdir)/sawfish/client.la
 
--- sawfish-1.0/configure.in~	Wed Sep  5 19:26:13 2001
+++ sawfish-1.0/configure.in	Wed Sep  5 19:42:07 2001
@@ -22,7 +22,6 @@
 
 AC_INIT(src/sawmill.h)
 AC_CONFIG_HEADER(config.h)
-AC_CONFIG_AUX_DIR(etc)
 
 dnl Release versioning info
 version="1.0"
@@ -66,12 +65,7 @@
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_PROG_GCC_TRADITIONAL
-
-dnl Workaround etc/install-sh not being found from subdirectories
-if test "$INSTALL" = "$ac_install_sh"; then
-  dnl Why is the backslash needed?
-  INSTALL='\${top_srcdir}/etc/install-sh -c'
-fi
+AC_PROG_LIBTOOL
 
 dnl Checks for libraries.
 AC_CHECK_LIB(socket, bind)
@@ -368,9 +362,6 @@
 AC_SUBST(localedir)
 AC_SUBST(HAVE_X11)
 AC_SUBST(LIBOBJS)
-AC_SUBST(CPPFLAGS)
-AC_SUBST(CFLAGS)
-AC_SUBST(LIBS)
 AC_SUBST(X_CFLAGS)
 AC_SUBST(X_LIBS)
 AC_SUBST(X_EXTRA_LIBS)
